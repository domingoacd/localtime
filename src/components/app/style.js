import styled from "styled-components";

export const AppContainer = styled.div`
  position: relative;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 2rem ${({ theme }) => theme.spacing.xPadding};
  overflow: hidden;
  background-size: cover;
  background-image: url(${(props) => props.bgImage});
`;
