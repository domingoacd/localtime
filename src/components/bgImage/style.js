import styled from "styled-components";

export const Image = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-size: cover;
  background-image: url(${(props) => props.url});
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
`;
