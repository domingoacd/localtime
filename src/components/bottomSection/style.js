import styled from "styled-components";

export const Section = styled.section`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  line-height: 0.9;
`;

export const WelcomeText = styled.p`
  font-size: ${({ theme }) => theme.fontSize.medium};
  text-transform: uppercase;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const LocationText = styled.p`
  font-size: ${({ theme }) => theme.fontSize.medium};
  font-weight: bold;
  text-transform: uppercase;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const LeftSide = styled.div``;

export const RightSide = styled.div``;
