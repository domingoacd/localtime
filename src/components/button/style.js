import styled from "styled-components";

export const Btn = styled.button`
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 0.5rem 0.7rem;
  text-transform: uppercase;
  border-radius: 5rem;
  border: none;
  cursor: pointer;
  color: ${({ theme }) => theme.day.color.secondary};
  background-color: ${({ theme }) => theme.day.color.primary};
`;

export const Arrow = styled.span`
  display: block;
  width: 1rem;
  height: 1rem;
  transform: ${(props) => (props.rotate ? "rotate(180deg)" : "rotate(0deg)")};
  transition: transform 0.5s;
  background-image: url(${(props) => props.image});
  background-size: 100% 100%;
`;
