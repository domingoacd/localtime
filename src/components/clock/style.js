import styled from "styled-components";

export const Time = styled.p`
  font-weight: bold;
  font-size: clamp(2rem, 25vw, 20rem);
  color: ${({ theme }) => theme.day.color.primary};
`;
