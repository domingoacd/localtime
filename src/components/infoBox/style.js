import styled from "styled-components";

export const Container = styled.div`
  position: absolute;
  bottom: 0;
  width: 100vw;
  display: flex;
  gap: 20vw;
  padding: 3rem;
  transform: translateY(100%);
  transition: transform 0.5s;
  margin: 0 -${({ theme }) => theme.spacing.xPadding};
  background-color: rgba(255, 255, 255, 0.7);
`;

export const Col = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5rem;
`;

export const InfoElement = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.4rem;
  line-height: 0.8;
`;

export const InfoName = styled.p`
  font-weight: bold;
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.fontSize.small};
  color: ${({ theme }) => theme.day.color.third};
`;

export const Info = styled.h3`
  font-weight: bold;
  font-size: ${({ theme }) => theme.fontSize.big};
  color: ${({ theme }) => theme.day.color.secondary};
`;
