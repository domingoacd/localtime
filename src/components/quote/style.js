import styled from "styled-components";

export const Blockquote = styled.blockquote`
  position: relative;
  display: flex;
  flex-direction: column;
  font-size: ${({ theme }) => theme.fontSize.small};
  visibility: ${(props) => (props.show ? "visible" : "hidden")};
  opacity: ${(props) => (props.show ? "1" : "0")};
  transition: visibility 0.5s, opacity 0.5s;
  color: ${({ theme }) => theme.day.color.primary};
`;

export const QuoteAuthor = styled.span`
  font-weight: bold;
  margin-top: 1rem;
`;
