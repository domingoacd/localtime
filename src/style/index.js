import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
* {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    font-family: 'Roboto', sans-serif;
}
`;

export const theme = {
  day: {
    color: {
      primary: "#fff",
      secondary: "#383838",
      third: "#696969",
    },
  },

  night: {
    color: {
      primary: "#fff",
      secondary: "#fff",
      third: "#bfbfbf",
    },
  },

  fontSize: {
    small: "1.1rem",
    medium: "1.4rem",
    big: "2rem",
    xbig: "30rem",
  },
};
