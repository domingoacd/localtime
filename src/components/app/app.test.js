import React from "react";
import renderer from "react-test-renderer";
import App from "../app";
import { ThemeProvider } from "styled-components";
import { theme } from "../../style";

test("Renders App component", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
