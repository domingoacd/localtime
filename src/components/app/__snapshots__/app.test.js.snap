// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App component 1`] = `
Array [
  <div
    className="sc-dlnjwi fDzMIF"
  >
    <div
      className="sc-hKFxyN jIwCHt"
    />
  </div>,
  <div
    className="sc-eCApnc jMzDgT"
  >
    <blockquote
      className="sc-bdnxRM gBQhwI"
    >
      "Donâ€™t postpone joy until you have learned all of your lessons. Joy is your lesson."
      <span
        className="sc-gtsrHT kyKqlw"
      >
        Alan Cohen
      </span>
    </blockquote>
    <section
      className="sc-fujyAs dPWxxo"
    >
      <div
        className="sc-iqAclL"
      >
        <p
          className="sc-pNWdM bfEEQN"
        >
          <img
            alt=""
            className="sc-kEqXSa ewoxua"
            src="moon.png"
          />
           GOOD NIGHT, THE TIME IS
        </p>
        <p
          className="sc-jSFjdj eUeLBm"
        >
          07:02
        </p>
        <p
          className="sc-jrsJWt bwYHFv"
        >
          IN CARACAS, VENEZUELA
        </p>
      </div>
      <button
        className="sc-gKAaRy iRJMvI"
      >
        Show more 
        <span
          className="sc-iCoGMd ewA-Dvh"
        />
      </button>
    </section>
    <div
      className="sc-crzoAE eNHenv"
    >
      <div
        className="sc-dIsUp ipoGrL"
      >
        <div
          className="sc-bqGGPW kqzvNC"
        >
          <p
            className="sc-ksluID ieOeuJ"
          >
            CURRENT TIMEZONE
          </p>
          <h3
            className="sc-hBMUJo bbVOhK"
          >
            Venezuelan VET
          </h3>
        </div>
        <div
          className="sc-bqGGPW kqzvNC"
        >
          <p
            className="sc-ksluID ieOeuJ"
          >
            DAY OF THE WEEK
          </p>
          <h3
            className="sc-hBMUJo bbVOhK"
          >
            1
          </h3>
        </div>
      </div>
      <div
        className="sc-dIsUp ipoGrL"
      >
        <div
          className="sc-bqGGPW kqzvNC"
        >
          <p
            className="sc-ksluID ieOeuJ"
          >
            DAY OF THE YEAR
          </p>
          <h3
            className="sc-hBMUJo bbVOhK"
          >
            125
          </h3>
        </div>
        <div
          className="sc-bqGGPW kqzvNC"
        >
          <p
            className="sc-ksluID ieOeuJ"
          >
            WEEK NUMBER
          </p>
          <h3
            className="sc-hBMUJo bbVOhK"
          >
            29
          </h3>
        </div>
      </div>
    </div>
  </div>,
]
`;
